File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/studentWeb/src/contracts/aiModelContract.ts
export interface AiModelContract {
  contractName: 'AiModel.v1'
  modelId: string
  displayName: string
  provider: string
  capabilities: string[]
  available: boolean
}

export interface AiModelRegistryContract {
  contractName: 'AiModelRegistry.v1'
  providerAvailable: boolean
  provider: string
  defaultMode: 'AUTO'
  models: AiModelContract[]
  warning: string | null
}

File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/backend/app/contracts/aiModelContract.py
from typing import Literal

from pydantic import BaseModel


class AiModelContract(BaseModel):
    contractName: Literal["AiModel.v1"] = "AiModel.v1"
    modelId: str
    displayName: str
    provider: str
    capabilities: list[str]
    available: bool


class AiModelRegistryContract(BaseModel):
    contractName: Literal["AiModelRegistry.v1"] = "AiModelRegistry.v1"
    providerAvailable: bool
    provider: str
    defaultMode: Literal["AUTO"] = "AUTO"
    models: list[AiModelContract]
    warning: str | None = None
